src/algebra/invertible.lean

```diff
@@ -104,15 +104,22 @@ def unit_of_invertible [monoid α] (a : α) [invertible a] : units α :=
 lemma is_unit_of_invertible [monoid α] (a : α) [invertible a] : is_unit a :=
 ⟨unit_of_invertible a, rfl⟩
 
+/-- Units are invertible in their associated monoid. -/
+instance units.invertible [monoid α] (u : units α) : invertible (u : α) :=
+{ inv_of := ↑(u⁻¹), inv_of_mul_self := u.inv_mul, mul_inv_of_self := u.mul_inv }
```

Comment:
I'm also a bit wary about coherence issues between this instance and others, especially `invertible_of_group`. Probably this one + `invertible_one` are fine since there is a cast in the conclusion. So e.g. we won't see the typeclass system reduce `↑(1 : units ℤ)` to `1 : ℤ`, right?

Reply:
Let's play it safe, I'll just back out the `instance`s for now

Reply:
Let's put a warning in the "implementation notes" above, so the next people to come across the file will know. I can't make a suggestion there, but how about something like:

````text
Since `invertible a` is not a `Prop` (but it is a `subsingleton`), we have to be careful about coherence issues:
we should avoid having multiple non-defeq instances for `invertible a` in the same context.
This file plays it safe and uses `def` rather than `instance` for most definitions,
users can choose which instances to use at the point of use.

For example, here's how you can use an `invertible 1` instance:
```lean
variables {α : Type*} [monoid α]

def something_that_needs_inverses (x : α) [invertible x] := sorry

section
local attribute [instance] invertible_one
def something_one := something_that_needs_inverses 1
end
```
````

Reply:
I don't fully buy that argument (and looking at the Zulip thread nor really did @digama0), so would prefer not to be responsible for PRing it.

Can you make a new PR adding that text?

Reply:
Created #8197 :+1: